test/filters/test_crop_hull.cpp

```diff
@@ -0,0 +1,297 @@
+      pcl::CropHull<pcl::PointXYZ> createDefaultCropHull (pcl::PointCloud<pcl::PointXYZ>::ConstPtr inputCloud) const
+      {
+        //pcl::CropHull<pcl::PointXYZ> cropHullFilter(true);
+        pcl::CropHull<pcl::PointXYZ> cropHullFilter;
+        pcl::ConvexHull<pcl::PointXYZ> convexHull;
+        convexHull.setDimension(3);
+        convexHull.setInputCloud(inputCloud);
+        pcl::PointCloud<pcl::PointXYZ>::Ptr hullCloudPtr(new pcl::PointCloud<pcl::PointXYZ>);
+        std::vector<pcl::Vertices> hullPolygons;
+        convexHull.reconstruct(*hullCloudPtr, hullPolygons);
+        cropHullFilter.setHullIndices(hullPolygons);
+        cropHullFilter.setHullCloud(hullCloudPtr);
+        cropHullFilter.setDim(CropHullTestTraits::getDim());
+        return cropHullFilter;
+      }
+
+      mutable std::mt19937 gen;
+      mutable std::uniform_real_distribution<float> rd;
+      pcl::PointCloud<pcl::PointXYZ> baseOffsetList;
+  };
+
+
+  struct CropHullTestTraits2d
+  {
+    static pcl::PointCloud<pcl::PointXYZ>::ConstPtr getInputCloud()
+    {
+      static pcl::PointCloud<pcl::PointXYZ>::Ptr inputCloud (new pcl::PointCloud<pcl::PointXYZ>);
+      if (inputCloud->empty()) {
+        for (const float i: {0.f, 1.f})
+          for (const float j: {0.f, 1.f})
+            for (const float k: {0.f, -0.1f})
+              inputCloud->emplace_back(i, j, k);
+      }
+      return inputCloud;
+    }
+    static int getDim() {
+      return 2;
+    }
+  };
+
+
+  struct CropHullTestTraits3d
+  {
+    static pcl::PointCloud<pcl::PointXYZ>::ConstPtr getInputCloud()
+    {
+      static pcl::PointCloud<pcl::PointXYZ>::Ptr inputCloud (new pcl::PointCloud<pcl::PointXYZ>);
+      if (inputCloud->empty()) {
+        for (const float i: {0.f, 1.f})
+          for (const float j: {0.f, 1.f})
+            for (const float k: {0.f, 1.f})
+              inputCloud->emplace_back(i, j, k);
+      }
+      return inputCloud;
+    }
+    static int getDim() {
+      return 3;
+    }
+  };
+}
+using CropHullTestTypes = ::testing::Types<CropHullTestTraits2d, CropHullTestTraits3d>;
+TYPED_TEST_SUITE(PCLCropHullTestFixture, CropHullTestTypes);
+
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// since test input cloud has same distribution for all dimensions, this test also check problem from issue #3960 //
+TYPED_TEST (PCLCropHullTestFixture, simple_test)
+{
+  for (auto & entry : this->data)
+  {
+    auto & cropHullFilter = entry.first;
+    for (TestData const & testData : entry.second)
+    {
+      cropHullFilter.setInputCloud(testData.inputCloud);
+      pcl::Indices filteredIndices;
+      cropHullFilter.filter(filteredIndices);
+      pcl::test::EXPECT_EQ_VECTORS(testData.insideIndices, filteredIndices);
+    }
+  }
+}
+
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// this test will pass only for 2d case //
```

Comment:
Is this a bug?

Reply:
This is bug, which was described in issue #1657

Reply:
If #3883 fixes this, then it's a go